### PR TITLE
Add basic support for PEP 695

### PR DESCRIPTION
**Pull request by zanieb**
Part of https://github.com/astral-sh/ruff/issues/5062

## Summary

Adds basic boilerplate support for PEP 695 without the expectation that we will actually _do_ anything with the contents of such expressions.

Bumps our RustPython/Parser fork to get support for PEP 695 as tracked by https://github.com/RustPython/Parser/issues/82

### CODE REVIEW

File: crates/ruff_python_ast/src/visitor.rs [hunk 1]
```diff
@@ -572,13 +572,7 @@ pub fn walk_expr<'a, V: Visitor<'a> + ?Sized>(visitor: &mut V, expr: &'a Expr) {
...
 }
 
-pub fn walk_constant<'a, V: Visitor<'a> + ?Sized>(visitor: &mut V, constant: &'a Constant) {
-    if let Constant::Tuple(constants) = constant {
-        for constant in constants {
-            visitor.visit_constant(constant);
-        }
-    }
-}
+pub fn walk_constant<'a, V: Visitor<'a> + ?Sized>(visitor: &mut V, constant: &'a Constant) {}
```
Comment (zanieb) [inline crates/ruff_python_ast/src/visitor.rs:575]: Should we remove all visiting of constants? As far as I can tell https://github.com/astral-sh/ruff/blob/22bcd0afb428a4ef4a9426ae6374693aa9bf8043/crates/ruff_python_ast/src/visitor.rs#L510 is unused now

Comment (zanieb) [reply crates/ruff_python_ast/src/visitor.rs:575]: See #5840

File: crates/ruff_python_ast/src/source_code/generator.rs [hunk 2]
```diff
@@ -538,6 +539,16 @@ impl<'a> Generator<'a> {
                     self.indent_depth = self.indent_depth.saturating_sub(1);
                 }
             }
+            Stmt::TypeAlias(ast::StmtTypeAlias { name, range: _range, type_params, value}) => {
+                self.p("type ");
+                self.unparse_expr(name, precedence::MAX);
+                for type_param in type_params {
+                    self.unparse_type_param(type_param);
+                    self.p(", ")
```
Comment (zanieb) [inline crates/ruff_python_ast/src/source_code/generator.rs:547]: Need to get a little more sophisticated in handling trailing comma, I presume.

File: crates/ruff_python_ast/src/helpers.rs [hunk 3]
```diff
@@ -265,6 +265,24 @@ where
...
+pub fn any_over_type_param<F>(type_param: &TypeParam, func: &F) -> bool
+where
+    F: Fn(&Expr) -> bool,
+{
+    match type_param {
+        TypeParam::TypeVar(ast::TypeParamTypeVar { bound, .. }) => {
+            bound.as_ref().map_or(false, |value| any_over_expr(value, func))
+        }
+        TypeParam::TypeVarTuple(ast::TypeParamTypeVarTuple { .. }) => {
+            false
```
Comment (zanieb) [inline crates/ruff_python_ast/src/helpers.rs:278]: Should `any_over_expr` visit names?

Comment (charliermarsh) [reply crates/ruff_python_ast/src/helpers.rs]: It should visit all expressions, so needs to recurse on anything that might contain an expression.

Comment (zanieb) [reply crates/ruff_python_ast/src/helpers.rs]: The names can't contain expressions, they're just containers for a string. It seems like a pain if you _were_ trying to visit parameter names here but I guess we can tackle that with another helper (or as it comes up).

**Pull request closed without merging**